cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterUtils.java

```diff
@@ -921,6 +923,28 @@ static int partitionBoundary(HttpServletRequest request, boolean isUpperBound) {
     return Integer.parseInt(boundaries[isUpperBound ? 1 : 0]);
   }
 
+  /**
+   * Get the {@link #NUM_BROKERS_TO_ADD} from the request.
+   *
+   * Default: -1
+   * @return The value of {@link #NUM_BROKERS_TO_ADD} parameter.
+   */
+  static int numBrokersToAdd(HttpServletRequest request) {
+    String parameterString = caseSensitiveParameterName(request.getParameterMap(), NUM_BROKERS_TO_ADD);
+    return parameterString == null ? -1 : Integer.parseInt(request.getParameter(parameterString));
+  }
+
+  /**
+   * Get the {@link #PARTITION_COUNT} from the request.
+   *
+   * Default: -1
+   * @return The value of {@link #PARTITION_COUNT} parameter.
+   */
+  static int partitionCount(HttpServletRequest request) {
+    String parameterString = caseSensitiveParameterName(request.getParameterMap(), PARTITION_COUNT);
+    return parameterString == null ? -1 : Integer.parseInt(request.getParameter(parameterString));
+  }
+
```

Comment:
1. If the response can be `null`, can we change the return type from `int` to `Integer` and return `null` as the default value? Would it make the response more clear?
2. Can we ensure that if the response is not `null`, then it must be a positive number? -- e.g. partition count or num brokers of `-3` would throw a `UserRequestException`.

Reply:
1. The response here shouldn't be null. It is set to -1 due to the `ProvisionRecommendation.DEFAULT_OPTIONAL_INT` being -1
2. Adding a check to make sure the response is a positive number outside of the default.